src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -88,6 +89,7 @@ public void setAddressBookFilePath(Path addressBookFilePath) {
     @Override
     public void setAddressBook(ReadOnlyAddressBook addressBook) {
         versionedAddressBook.resetData(addressBook);
+        undoableCommand = "clear";
```

Comment:
Maybe should this clear the undoableCommand list instead, when the data is reset? If not, will the list ever be cleared?

Reply:
Think it's good to go